### DISCUSSION TITLE
The Mysterious Case of the Grayed Button

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#8464] [Q&A]
I published my app yesterday with MacOS support.
https://github.com/mysteryx93/HanumanInstituteApps/releases/tag/v2.1.1

There is an information button. On Linux, it works. On Windows, it's sometimes grayed out. On MacOS, it never works.

I cannot reproduce the problem in Release builds. Only after publishing and trimming.

After investigation, it only happens when these 2 conditions happen:

1. I'm initializing the View and the ViewModel/ViewModelLocator in parallel for faster startup.
https://github.com/mysteryx93/HanumanInstituteApps/blob/master/Common.Avalonia.App/Utilities/CommonApplication.cs#L43=

2. I trim the assembly

If I change either of these conditions, the problem goes away (at least on Windows, haven't yet confirmed on MacOS)

The button itself is bound to ShowAbout command and has no reason to be grayed out.
https://github.com/mysteryx93/HanumanInstituteApps/blob/master/Converter432hz/ViewModels/MainViewModel.cs#L282=

So..... what's going on here? Can I initialize the View and ViewModel/ViewModelLocator in parallel or I have to drop that idea?

**Comment by maxkatz6** [comment]
I feel like aggressive trimmer might have deleted view model from assembly, and data context was left null. 
See https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options#trimming-granularity
`Enable member-level trimming, which removes unused members from types. This is the default in .NET 6+.`

You can either change your code to implicitly create the view model type, or disable trimming for the assembly, or add some attributes https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming#resolve-trim-warnings.

**Reply by maxkatz6** [reply]
But I can't explain why it's platform specific.

**Reply by mysteryx93** [reply]
DataContext is not null because everything else works perfectly fine.

**Reply by mysteryx93** [reply]
Adding `<TrimMode>copyused</TrimMode>` to the project seems to solve the problem. It doesn't impact the output size.
